### PR TITLE
Storage package

### PR DESCRIPTION
**Pull request by Pantani**
Create a generic package to handle database 

- Create a new generic Redis package
- Add Postgres support

### CODE REVIEW

File: pkg/storage/sql/postgres.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		return storage.ErrNotFound
+	}
+	err := db.client.Save(value).Error
+	if err != nil {
+		return errors.E(err, storage.ErrNotUpdated)
+	}
+	return nil
+}
+
+func (db *PgSql) Add(value interface{}) error {
```
Comment (vikmeup) [inline pkg/storage/sql/postgres.go]: do you need to specify table name?  or how does it work?

File: pkg/storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+var Redis redis.Redis
+var Postgres sql.PgSql
+
+func InitDatabases() {
+	err := Redis.Init(viper.GetString("observer.redis"))
+	if err != nil {
+		logger.Fatal(er.E(err), "Cannot connect to Redis")
+	}
```
Comment (vikmeup) [inline pkg/storage/storage.go]: can we potentially use just one database? because if one of the databases is down - blockatlas will fail

Comment (Pantani) [reply pkg/storage/storage.go]: I like Redis to storage the block numbers, is quickly and don't need a lot of processing
But we can keep only one database to don't make the infra more complex

File: pkg/storage/sql/postgres.go (hunk 1 above)
Comment (Pantani) [reply pkg/storage/sql/postgres.go]: I did use Gorm, it's a Golang ORM. It automatically creates my databases based on the structs

https://gorm.io/

Comment (Pantani) [reply pkg/storage/sql/postgres.go]: I going start to implement and you going to see is easier to use

**Comment by Pantani** [comment]
> Are we going to use MySQL / PQ at the same time?

only Postgres, but to change the SQL provides is easily just one line

**Comment by Pantani** [comment]
> @Pantani do you use Pg as NoSql db?

We as use as SQL (Relational) db

**Comment by kolya182** [comment]
@Pantani Could you justify use of PG next time when making such radical changes

**Comment by Pantani** [comment]
> @Pantani Could you justify use of PG next time when making such radical changes

Yes.
This PR don't make changes, only introduce a new package to handle databases. The database changes going to be in another PR. We need a better database with queries to handle the xpub things. I going to explain better and create a readme in the next PR

**Pull request merged**